### ISSUE TITLE
Dynamo chokes on nontrivial attribute access on intermediate DTensors

### ISSUE THREAD

**Issue by bdhirsh** [pytorch/pytorch#119918]
repro example:
```
+    def test_dtensor_attribute_access_on_intermediate(self):
+        mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+
+        def fn(x):
+            tmp = x * 2
+            if tmp.placements[0].is_shard():
+                return tmp._local_tensor + 2
+            else:
+                return tmp._local_tensor + 3
+
+        x = DTensor.from_local(torch.ones(4), mesh, [Shard(0)], run_check=False)
+        ref = fn(x)
+
+        opt_fn = torch.compile(fn, backend="aot_eager", fullgraph=True)
+        res = opt_fn(x)
+        self.assertEqual(res, ref)
+
```

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @ezyang @msaroufim @anijain2305 @zou3519

**Comment by bdhirsh** [comment]
Fixed by https://github.com/pytorch/pytorch/pull/118209

**Comment by masnesral** [comment]
> Fixed by #118209

@bdhirsh I'm attempting my oncall duties at issue cleanup. If this is fixed, then do we close? Or maybe assign to you to close when you're satisfied the fix is good.. or do we add "triaged" to take it out of our queue?

**Comment by Chillee** [comment]
Closing because it's fixed by #118209. @bdhirsh
